### ISSUE TITLE
libjpeg

### ISSUE THREAD

**Issue by ZYX0106** [uber-research/jpeg2dct#7]
Which version of libjpeg have you used in this paper?

**Comment by HoracceFeng** [comment]
Hi @ZYX0106 @uber-research, I am also facing the libjpeg version issue right now? My error msg is 'Wrong JPEG library version: library is 90, caller expect 62'. But actually I have installed the version62 and have the 'libjpeg.so.62'.

Anyone can help me?

**Comment by Zdafeng** [comment]
> Hi @ZYX0106 @uber-research, I am also facing the libjpeg version issue right now? My error msg is 'Wrong JPEG library version: library is 90, caller expect 62'. But actually I have installed the version62 and have the 'libjpeg.so.62'.
> 
> Anyone can help me?

Have you solved it？

**Comment by HoracceFeng** [comment]
Yes. To solve this problem, you should make sure you have correctly linked the libjpeg.62.so before you install jpeg2dct.

In my case, I install libjpeg62 under `/usr/local/lib`. But I find out that in the pip rootdir, I have the libjpeg90, so when I run the `pip install jpeg2dct`, the compiler will search pip rootdir first and linked the libjpeg90 but not libjpeg62.

It is easy to check if you have this problem. Just find out the .so file of jpeg2dct.common and use ldd to check the dependencies.

If you have the same issue like me, the easiest way to fix this is to rename the libjpeg90, then pip install jpeg2dct, after that, you can change the name back to libjpeg90. (I suggest you to do so, because pil and cv2 will use the libjpeg90)

**Comment by Zdafeng** [comment]
> Yes. To solve this problem, you should make sure you have correctly linked the libjpeg.62.so before you install jpeg2dct.
> 
> In my case, I install libjpeg62 under `/usr/local/lib`. But I find out that in the pip rootdir, I have the libjpeg90, so when I run the `pip install jpeg2dct`, the compiler will search pip rootdir first and linked the libjpeg90 but not libjpeg62.
> 
> It is easy to check if you have this problem. Just find out the .so file of jpeg2dct.common and use ldd to check the dependencies.
> 
> If you have the same issue like me, the easiest way to fix this is to rename the libjpeg90, then pip install jpeg2dct, after that, you can change the name back to libjpeg90. (I suggest you to do so, because pil and cv2 will use the libjpeg90)
> 
> 

Best

**Comment by sketchplanet** [comment]
> Yes. To solve this problem, you should make sure you have correctly linked the libjpeg.62.so before you install jpeg2dct.
> 
> In my case, I install libjpeg62 under `/usr/local/lib`. But I find out that in the pip rootdir, I have the libjpeg90, so when I run the `pip install jpeg2dct`, the compiler will search pip rootdir first and linked the libjpeg90 but not libjpeg62.
> 
> It is easy to check if you have this problem. Just find out the .so file of jpeg2dct.common and use ldd to check the dependencies.
> 
> If you have the same issue like me, the easiest way to fix this is to rename the libjpeg90, then pip install jpeg2dct, after that, you can change the name back to libjpeg90. (I suggest you to do so, because pil and cv2 will use the libjpeg90)

Hello when i rename the libjpeg.so.9 then pip install the jpeg2dct I encounter the error that OSError：libjpeg.so.9 cannot open  shared object file.

**Comment by Jinhui-Qin** [comment]
> > Yes. To solve this problem, you should make sure you have correctly linked the libjpeg.62.so before you install jpeg2dct.
> > In my case, I install libjpeg62 under `/usr/local/lib`. But I find out that in the pip rootdir, I have the libjpeg90, so when I run the `pip install jpeg2dct`, the compiler will search pip rootdir first and linked the libjpeg90 but not libjpeg62.
> > It is easy to check if you have this problem. Just find out the .so file of jpeg2dct.common and use ldd to check the dependencies.
> > If you have the same issue like me, the easiest way to fix this is to rename the libjpeg90, then pip install jpeg2dct, after that, you can change the name back to libjpeg90. (I suggest you to do so, because pil and cv2 will use the libjpeg90)
> 
> Hello when i rename the libjpeg.so.9 then pip install the jpeg2dct I encounter the error that OSError：libjpeg.so.9 cannot open shared object file.

Hi,I also get this issue, have you fixed it? please help me

**Comment by Jinhui-Qin** [comment]
> > Yes. To solve this problem, you should make sure you have correctly linked the libjpeg.62.so before you install jpeg2dct.
> > In my case, I install libjpeg62 under `/usr/local/lib`. But I find out that in the pip rootdir, I have the libjpeg90, so when I run the `pip install jpeg2dct`, the compiler will search pip rootdir first and linked the libjpeg90 but not libjpeg62.
> > It is easy to check if you have this problem. Just find out the .so file of jpeg2dct.common and use ldd to check the dependencies.
> > If you have the same issue like me, the easiest way to fix this is to rename the libjpeg90, then pip install jpeg2dct, after that, you can change the name back to libjpeg90. (I suggest you to do so, because pil and cv2 will use the libjpeg90)
> 
> Hello when i rename the libjpeg.so.9 then pip install the jpeg2dct I encounter the error that OSError：libjpeg.so.9 cannot open shared object file.

Hi, I install jpeg2dct from sources and make it. First, I find the location jpeg90' installing, then install jpeg62 into the same location and uninstall jpeg2dct, finally install jpeg2dct from sources.

**Comment by Tonyboy999** [comment]
> Yes. To solve this problem, you should make sure you have correctly linked the libjpeg.62.so before you install jpeg2dct.
> 
> In my case, I install libjpeg62 under `/usr/local/lib`. But I find out that in the pip rootdir, I have the libjpeg90, so when I run the `pip install jpeg2dct`, the compiler will search pip rootdir first and linked the libjpeg90 but not libjpeg62.
> 
> It is easy to check if you have this problem. Just find out the .so file of jpeg2dct.common and use ldd to check the dependencies.
> 
> If you have the same issue like me, the easiest way to fix this is to rename the libjpeg90, then pip install jpeg2dct, after that, you can change the name back to libjpeg90. (I suggest you to do so, because pil and cv2 will use the libjpeg90)

Would you describe `rename libjpeg90` in detail? I don't know what this means
